engine/src/flutter/shell/common/shell.cc

```diff
@@ -2361,12 +2407,39 @@ Shell::GetConcurrentWorkerTaskRunner() const {
   return vm_->GetConcurrentWorkerTaskRunner();
 }
 
-DlISize Shell::ExpectedFrameSize(int64_t view_id) {
-  auto found = expected_frame_sizes_.find(view_id);
-  if (found == expected_frame_sizes_.end()) {
-    return DlISize();
+BoxConstraints Shell::ExpectedFrameSize(int64_t view_id) {
+  auto found = expected_frame_constraints_.find(view_id);
+
+  if (found == expected_frame_constraints_.end()) {
+    return {};
   }
+
   return found->second;
 }
 
+bool Shell::IsSizeWithinConstraints(const DlISize& size,
```

Comment:
Can we move this to `BoxConstraints` and call this `IsSatisfiedBy`?

This would match Dart's `BoxConstraints.isSatisfiedBy`: https://api.flutter.dev/flutter/rendering/BoxConstraints-class.html

I expect you'll need to map from `DlISize` to `DlSize`, but I think that's OK.